### ISSUE TITLE
`&&=` and `||=` operators

### ISSUE THREAD

**Issue by lrhn** [dart-lang/language#122]
Dart does not currently support using the `&&` and `||` operators in assignments.
It allows other non-short-circuiting binary operators, and it allows `??=`, which is short-circuiting.
Users have expressed surprise that those two are not supported.

This is a solution to https://github.com/dart-lang/language/issues/23.

See https://dart-review.googlesource.com/c/sdk/+/43500/ for a specification.

**Comment by har79** [comment]
Dupe of https://github.com/dart-lang/language/issues/23?

**Comment by kevmoo** [comment]
Indeed a dupe of #23 – CC @mit-mit and @eernstg

**Comment by mit-mit** [comment]
Not really; #23 is a "language request" (aka a "problem"), the present bug is a "language feature" i.e. a possible solution to the former request/problem. For more details, see https://github.com/dart-lang/language/blob/master/doc/life_of_a_language_feature.md

I have clarified this in the top-comment.

**Comment by kevmoo** [comment]
@mit-mit Noticed the labels, too.

**Comment by munificent** [comment]
Closing in favor of #23. I don't think it's super helpful to have both issues around, especially for such a small feature.
